src/pages/images/actions/UploadInstanceBtn.tsx

```diff
@@ -0,0 +1,252 @@
+      void formik.setFieldTouched("name", true, true);
+      if (!formik.errors.name) {
+        formik.setFieldError("name", undefined);
+      }
+    }
+  };
+
+  const handleCloseModal = useCallback(() => {
+    uploadAbort?.abort();
+    setIsUploading(false);
+    setUploadState(null);
+    setUploadAbort(null);
+    formik.resetForm();
+    closePortal();
+  }, [uploadAbort]);
+
+  const onSuccess = (instanceName: string) => {
+    const message = (
+      <>
+        Created Instance <strong>{instanceName}</strong>.
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: () =>
+          navigate(
+            `/ui/project/${project?.name}/instance/${instanceName}/configuration`,
+          ),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const onFailure = (msg: string) => {
+    toastNotify.failure("Instance creation failed.", new Error(msg));
+  };
+
+  const onFinish = () => {
+    void queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.instances;
+      },
+    });
+  };
+
+  const handleUpload = () => {
+    if (!formik.values.instanceFile) {
+      return;
+    }
+
+    setIsUploading(true);
+    const uploadController = new AbortController();
+    setUploadAbort(uploadController);
+
+    const instanceName = formik.values.name;
+    void uploadInstance(
+      formik.values.instanceFile,
+      instanceName,
+      project?.name,
+      formik.values.pool,
+      setUploadState,
+      uploadController,
+    )
+      .then((operation) => {
+        toastNotify.info(
+          <>
+            Upload completed. Now creating instance{" "}
+            <strong>{instanceName}</strong>.
+          </>,
+        );
+
+        eventQueue.set(
+          operation.metadata.id,
+          () => onSuccess(instanceName),
+          onFailure,
+          onFinish,
+        );
+      })
+      .catch((e: AxiosError<LxdSyncResponse<null>>) => {
+        const error = new Error(e.response?.data.error);
+        toastNotify.failure("Instance upload failed", error);
+      })
+      .finally(() => {
+        handleCloseModal();
+        navigate(`/ui/project/${project?.name}/instances`);
+      });
+  };
+
+  const formik = useFormik<UploadInstanceFormValues>({
+    initialValues: {
+      name: "",
+      pool: "",
+      instanceFile: null,
+    },
+    validateOnMount: true,
+    validationSchema: Yup.object().shape({
+      name: instanceNameValidation(
+        project?.name || "",
+        instanceNameAbort,
+      ).optional(),
+    }),
+    onSubmit: handleUpload,
+  });
+
+  return (
+    <>
+      <Button onClick={openPortal} type="button" id="upload-instance-file">
+        <span>Import from file</span>
+      </Button>
+      {isOpen && (
+        <Portal>
+          <Modal
+            close={handleCloseModal}
+            className="upload-instance-modal"
+            title="Upload instance from file"
```

Comment:
```suggestion
            title="Import instance from file"
```
Should this match the button triggering it more closely?

Reply:
sounds good :+1: